Removed empty callbacks in controllers and updated documentation (#685) (#731)
ros-controls/ros2_control_demos: example_7/doc/userdoc.rst

The following code blocks will explain the requirements for writing a new controller.

The controller plugin for the tutorial robot is a class called ``RobotController`` that inherits from  ``controller_interface::ControllerInterface``. The ``RobotController`` must implement the following public methods:

1. ``command_interface_configuration``
2. ``state_interface_configuration``
3. ``update``

> The following methods are `managed node <https://design.ros2.org/articles/node_lifecycle.html>`__  transitions callbacks. These overrides are optional and only the ``on_configure``, ``on_activate`` and ``on_deactivate`` have been used in this example.The ``on_cleanup`` and ``on_shutdown`` methods are called when the controller's lifecycle node transitions to the shutdown state. They should handle memory deallocation and general cleanup.The ``on_error`` method is invoked if the managed node encounters a failure during a state transition.

Reviewer: ```suggestion
The following methods are `managed node <https://design.ros2.org/articles/node_lifecycle.html>`__  transitions callbacks. These overrides are optional and only the ``on_configure``, ``on_activate`` and ``on_deactivate`` have been used in this example.
The ``on_cleanup`` and ``on_shutdown`` methods are called when the controller's lifecycle node transitions to the shutdown state. They should handle memory deallocation and general cleanup. The ``on_error`` method is invoked if the managed node encounters a failure during a state transition.
```

The following code blocks will explain the requirements for writing a new controller.

> The controller plugin for the tutorial robot is a class called ``RobotController`` that inherits from  ``controller_interface::ControllerInterface``. The ``RobotController`` must implement nine public methods. The last six are `managed node <https://design.ros2.org/articles/node_lifecycle.html>`__  transitions callbacks. However there are additional optional overrides.

Reviewer: ```suggestion
The controller plugin for the tutorial robot is a class called ``RobotController`` that inherits from  ``controller_interface::ControllerInterface``. The ``RobotController`` must implement the following public methods:
```

8. ``on_error``

> 9. ``on_shutdown``

Reviewer: move literals 4-9 below the next paragraph

Author: Done in the next commit.

Note:

> These overrides are optional and only the ``on_configure``,``on_activate`` and ``on_deactivate`` have been used in this example.

Reviewer: ```suggestion
These overrides are optional and only the ``on_configure``, ``on_activate`` and ``on_deactivate`` have been used in this example.
```

> Note:
> These overrides are optional and only the ``on_configure``,``on_activate`` and ``on_deactivate`` have been used in this example.

Reviewer: ```suggestion
The following methods are `managed node <https://design.ros2.org/articles/node_lifecycle.html>`__  transitions callbacks. These overrides are optional and only the ``on_configure``,``on_activate`` and ``on_deactivate`` have been used in this example.
```

      return CallbackReturn::SUCCESS;
  }

> The ``on_cleanup`` and ``on_shutdown`` are called when the controller's lifecycle node is transitioning to shutting down. Freeing any allocated memory and general cleanup should be done in these methods.

Reviewer: Move the on_cleanup, on_shutdown, and on_error description (without the code) to the

> The following methods are

paragraph above

Author: Done in the next commit.
